shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -76,238 +72,268 @@
+   * @param appBundlePath the path to the app bundle which contains the Dart app to execute
+   * @param dartEntrypoint the name of the initial Dart method to invoke, defaults to "main"
+   * @param flutterShellArgs any special configuration arguments for the Flutter engine
+   *
+   * @return Bundle of arguments that configure a {@link FlutterFragment}
+   */
+  public static Bundle createArgsBundle(boolean isSplashScreenDesired,
+                                        @Nullable String initialRoute,
+                                        @Nullable String appBundlePath,
+                                        @Nullable String dartEntrypoint,
+                                        @Nullable FlutterShellArgs flutterShellArgs) {
     Bundle args = new Bundle();
     args.putBoolean(ARG_IS_SPLASH_SCREEN_DESIRED, isSplashScreenDesired);
     args.putString(ARG_INITIAL_ROUTE, initialRoute);
     args.putString(ARG_APP_BUNDLE_PATH, appBundlePath);
-    frag.setArguments(args);
-
-    return frag;
+    args.putString(ARG_DART_ENTRYPOINT, dartEntrypoint);
+    if (null != flutterShellArgs) {
+      args.putStringArray(ARG_FLUTTER_INITIALIZATION_ARGS, flutterShellArgs.toArray());
+    }
+    return args;
   }
 
   private FlutterEngine flutterEngine;
   private FrameLayout container;
   private FlutterView flutterView;
-  private PlatformPlugin platformPlugin;
-  private BasicMessageChannel<String> mFlutterLifecycleChannel;
-  private BasicMessageChannel<Object> mFlutterSystemChannel;
-  private MethodChannel mFlutterNavigationChannel;
   private View launchView;
 
-  @Override
-  public View onCreateView(LayoutInflater inflater, ViewGroup parent, Bundle savedInstanceState) {
-    Log.e(TAG, "onCreateView()");
-    createLayout();
-    // TODO: Should we start running the FlutterView here or when attached to window? It was being done here
-    //       in the Activity, but maybe that was a problem to begin with?
-    flutterView.attachToFlutterRenderer(flutterEngine.getRenderer(), flutterEngine.getDartExecutor());
-    doInitialFlutterViewRun();
+  private PlatformPlugin platformPlugin;
 
-    return container;
+  public FlutterFragment() {
+    // Ensure that we at least have an empty Bundle of arguments so that we don't
+    // need to continually check for null arguments before grabbing one.
+    setArguments(new Bundle());
   }
 
   @Override
-  public void onStart() {
-    super.onStart();
-    Log.d(TAG, "onStart()");
-    mFlutterLifecycleChannel.send("AppLifecycleState.inactive");
-  }
+  public void onAttach(Activity activity) {
+    super.onAttach(activity);
 
-  public void onPostResume() {
-    Log.d(TAG, "onPostResume()");
-    flutterView.updateAccessibilityFeatures();
-    platformPlugin.onPostResume();
-    mFlutterLifecycleChannel.send("AppLifecycleState.resumed");
-  }
+    // TODO(mattcarroll): I think the build system is linking the wrong Fragment API. It says that
+    //                    getContext() cannot be found...
```

Comment:
Document the reason. E.g. is this a result of the buildroot's current Android SDK level, or some other reason?

Reply:
clarified